sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/BlobAsyncClientBase.java

```diff
@@ -545,27 +546,39 @@ public boolean isSnapshot() {
      * @throws UncheckedIOException If an I/O error occurs.
      */
-    public Mono<Void> downloadToFile(String filePath, BlobRange range, Integer blockSize,
-        ReliableDownloadOptions options, BlobAccessConditions accessConditions, boolean rangeGetContentMD5) {
-        return withContext(context -> downloadToFile(filePath, range, blockSize, options, accessConditions,
+    public Mono<Response<BlobProperties>> downloadToFileWithResponse(String filePath, BlobRange range,
+            Integer blockSize, ReliableDownloadOptions options, BlobAccessConditions accessConditions,
+            boolean rangeGetContentMD5) {
+        return withContext(context -> downloadToFileWithResponse(filePath, range, blockSize, options, accessConditions,
             rangeGetContentMD5, context));
     }
 
-    Mono<Void> downloadToFile(String filePath, BlobRange range, Integer blockSize, ReliableDownloadOptions options,
-        BlobAccessConditions accessConditions, boolean rangeGetContentMD5, Context context) {
+    Mono<Response<BlobProperties>> downloadToFileWithResponse(String filePath, BlobRange range, Integer blockSize,
+        ReliableDownloadOptions options, BlobAccessConditions accessConditions, boolean rangeGetContentMD5,
+        Context context) {
         if (blockSize != null) {
             Utility.assertInBounds("blockSize", blockSize, 0, BLOB_MAX_DOWNLOAD_BLOCK_SIZE);
         }
 
         return Mono.using(() -> downloadToFileResourceSupplier(filePath),
-            channel -> Mono.justOrEmpty(range)
-                .switchIfEmpty(getFullBlobRange(accessConditions))
-                .flatMapMany(rg -> Flux.fromIterable(sliceBlobRange(rg, blockSize)))
-                .flatMap(chunk -> this.download(chunk, accessConditions, rangeGetContentMD5, context)
-                    .subscribeOn(Schedulers.elastic())
-                    .flatMap(dar -> FluxUtil.writeFile(dar.body(options), channel,
-                        chunk.getOffset() - (range == null ? 0 : range.getOffset()))))
-                .then(), this::downloadToFileCleanup);
+            channel -> getPropertiesWithResponse(accessConditions).flatMap(response -> processInRange(channel, response,
+                range, blockSize, options, accessConditions, rangeGetContentMD5, context)),
+            this::downloadToFileCleanup);
+
+    }
+
+    private Mono<Response<BlobProperties>> processInRange(AsynchronousFileChannel channel,
+                Response<BlobProperties> blobPropertiesResponse, BlobRange range, Integer blockSize,
+                ReliableDownloadOptions options, BlobAccessConditions accessConditions, boolean rangeGetContentMD5,
+                Context context) {
+        return Mono.justOrEmpty(range).switchIfEmpty(Mono.just(new BlobRange(0,
+            blobPropertiesResponse.getValue().getBlobSize()))).flatMapMany(rg ->
+            Flux.fromIterable(sliceBlobRange(rg, blockSize)))
```

Comment:
In general, new line splits are preferred on the higher level rather than this lower level break.

```java
return Mono.justOrEmpty(range)
    .switchIfEmpty(Mono.just(new BlobRange(0,  blobPropertiesResponse.getValue().getBlobSize())))
    .flatMapMany(rg -> Flux.fromIterable(sliceBlobRange(rg, blockSize)))
    .flatMap(chunk -> ...
```